exonum/src/node/mod.rs

```diff
@@ -605,9 +607,15 @@ impl NodeHandler {
     }
 
     /// Performs connection to the specified network address.
-    pub fn connect(&mut self, address: &SocketAddr) {
+    pub fn connect(&mut self, address: &str) {
         let connect = self.state.our_connect_message().clone();
-        self.send_to_addr(address, connect.raw());
+        let resolved = self.state.resolve_and_cache_peer_address(address);
+        if let Some(socket) = resolved {
+            self.send_to_addr(&socket, connect.raw());
+        } else {
+            info!("Failed to resolve hostname of peer: {}", address);
+            return;
```

Comment:
This does not look like a fatal error to me. We may try other addresses that we got from the resolver before giving up.

Also, I believe the log line should use `error!` or `warn!` instead of `info!`. And is it really okay to just log and ignore the error here? Do the callers expect the connection to happen asynchronously? Can they reasonably handle a connection error?

Reply:
This else handles the case when we have 0 IPs from resolving address, we cannot try another for that reason.